### ISSUE TITLE
Update package to address deprecation warnings

### ISSUE THREAD

**Issue by ethanwhite** [weecology/DeepForest#625]
We have a growing number of deprecation warnings accumulating. Below are a number of these warnings that are produced by the current test suite. These aren't all necessarily from DeepForest directly (they may be upstream), but we should start finding the ones that are in DeepForest and update to get ahead of these changes. For folks looking for starter issues a great thing to do is to pick a single warning below, find one of our tests that produces the deprecation warning, and then see if the warning is triggered directly by the code run by that test or if it is upstream. If it is in the DeepForest code then work on a PR to fix just that warning and run the whole test suite to make sure that your fixes catches all instances of it in the code base.

- [x] The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.
- [x] Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=FasterRCNN_ResNet50_FPN_Weights.COCO_V1`. You can also use `weights=FasterRCNN_ResNet50_FPN_Weights.DEFAULT` to get the most up-to-date weights.
- [x] datetime.datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.fromtimestamp(timestamp, datetime.UTC).
- [x] Expected None, but tests/test_FasterRCNN.py::test_retinanet returned <deepforest.models.FasterRCNN.Model object at 0x7f91cba5d940>, which will be an error in a future version of pytest.  Did you mean to use `assert` instead of `return`?
- [x] The verbose parameter is deprecated. Please use get_last_lr() to access the learning rate.
- [ ] The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.
- [x] Type google._upb._message.MessageMapContainer uses PyType_Spec with a metaclass that has custom tp_new. This is deprecated and will no longer be allowed in Python 3.14.
- [x] Type google._upb._message.ScalarMapContainer uses PyType_Spec with a metaclass that has custom tp_new. This is deprecated and will no longer be allowed in Python 3.14.
- [x] The `op` parameter is deprecated and will be removed in a future release. Please use the `predicate` parameter instead.
- [x] ~"Directly specifying the num_classes arg in deepforest.main will be deprecated in 2.0 in favor of config_args. Use main.deepforest(config_args={'num_classes':value})" [main.py line 66](https://github.com/weecology/DeepForest/blob/a4c12a606c295bb5a32e63d1d156ea2b8a52c002/deepforest/main.py#L64)~

If you are working on one of these please drop a note in this issue with which warning you are working on so that folks don't duplicate effort.

EDIT: Removed the last item since we're generating it intentionally

**Comment by Mu-Magdy** [comment]
That is a good issue I will work on the first warning and make a PR
* _The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead._

**Comment by Mu-Magdy** [comment]
As a result of fixing the first deprecation warning I think it solved the second warning as they are related

**Comment by Mu-Magdy** [comment]
I will work on the last task 
* _The op parameter is deprecated and will be removed in a future release. Please use the predicate parameter instead._

**Comment by Om-Doiphode** [comment]
I have worked on the fifth issue:

`The verbose parameter is deprecated. Please use get_last_lr() to access the learning rate.`

I have issued a pull request, please review it. Thanks!

**Comment by ire-mide1** [comment]
I will work on issue 4

_Expected None, but tests/test_FasterRCNN.py::test_retinanet returned <deepforest.models.FasterRCNN.Model object at 0x7f91cba5d940>, which will be an error in a future version of pytest. Did you mean to use assert instead of return?_

**Comment by ethanwhite** [comment]
With #738 and the early work we've cleaned up almost all of these and I don't ?think? we care about the FutureWarning from Pandas, so I'm going to go ahead and close. We should continue the more general work of trying to minimize the number of warnings in our tests, but it's not a high priority at the moment.
